upgrades/upgradevalidation/validation.go

```diff
@@ -249,6 +249,41 @@ func checkForCharmStoreCharms(_ string, _ StatePool, st State, _ Model) (*Blocke
 	return nil, nil
 }
 
+func checkForCharmsWithNoManifest(_ string, _ StatePool, st State, _ Model) (*Blocker, error) {
+	curls, err := st.AllCharmURLs()
+	if errors.Is(err, errors.NotFound) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	result := set.NewStrings()
+	for _, curl := range curls {
+		if curl == nil {
+			return nil, errors.New("malformed charm in database with no URL")
+		}
+		ch, err := st.Charm(*curl)
+		if err != nil {
+			return nil, errors.Annotatef(err, "getting charm %q", *curl)
+		}
```

Comment:
We need to skip charms that aren't uploaded or are placeholders. They potentially won't have manifests because they're not filled in correctly.

I wonder if we should allow migrations with charms that are not uploaded or are placeholders. It would potentially make the migrations more robust.

Reply:
Let's discuss in the go around.

Reply:
Discuessed yesterday is go round. This could be an issue for charms that have no units, but if a charm has units and it is pending/uploading then the unit will not be idle and migration will be blocked.